cocotb/scheduler.py

```diff
@@ -251,6 +251,8 @@ def __init__(self):
         self._write_coro_inst = None
         self._writes_pending = Event()
 
+        self._simulator_clocks = set()
```

Comment:
Hmmm, maybe we need a task-like object to represent an independent external process so we don't have to special case every time we add a new feature to C. Could this be modeled with `cocotb.external`?

Reply:
I'll look at `external`, but I'm not sure there's a way to signal it in order to have it stop the clock. We're missing something like #1349 for resource cleanup.

Reply:
Right... That's why I mentioned task-like. Something we can add to the scheduler that looks like a task and waits on a trigger that will never fire (`InfinityTrigger`?), that is `.kill`ed at the end of the test. Tasks don't have a well-defined interface according to how the scheduler uses them, do they?

Reply:
So we can support the current `Clock` interface by using `fork(SimClock(...).start())` like we do now.

Or we can do something completely different! I'm just throwing out ideas, it's your effort. I'm just here to make sure you dot your ı's, errr... i's.

Reply:
There's also no way currently to notify the scheduler or any task that a clock using `cycles` has finished toggling and stopped.

Reply:
Well it is cleaned up by the test end. Users have no way of knowing when the clock stopped either, which is the bigger issue I'd think.

Reply:
What about:
```py3
clock_coro = cocotb.fork(Clock(100, units='ns').start(cycles=15))
await Join(clock_coro)
```
Someone somewhere might be using that.

Reply:
Hmmmmm... The trigger would have to be set in C++.

I definitely don't think it's a good idea right now. Just getting a feel for what would have to be added to make that work. I have a feeling this isn't the first of these types of changes and having that functionality there would make adding similar features much easier in the future.

Reply:
Something like a `GPIEvent` trigger. It should be safe to call `Scheduler.react` (with GIL), as the C++ callback is just a timed callback like `Timer`.

That can wait for when `Clock` is moved to use this as a backend.
If we make that opt-in, we could probably get away with not supporting this at all and force users to poll `is_running` manually if they want to know when the clock stops.